File: ChallengeStefaniniGroup.API/ChallengeStefaniniGroup.Application/TaskService.cs
using ChallengeStefaniniGroup.Data.Data;
using ChallengeStefaniniGroup.Data.Repository.TaskRepository;
using MongoDB.Bson;
namespace ChallengeStefaniniGroup.Application
{
    public class TaskService : ITaskService
    {
        protected TaskRepository TaskRepository { get; init; }
        public TaskService(DataContext context) => TaskRepository = new TaskRepository(context);
        public async Task<IEnumerable<Domain.Entities.Task>> GetAllTasks() => await TaskRepository.GetAll();
        public async Task<Domain.Entities.Task?> GetTaskById(ObjectId id) => await TaskRepository.GetById(id);
        public async Task AddTask(Domain.Entities.Task newTask) => await TaskRepository.Add(newTask);
        public async Task DeleteTask(ObjectId id) => await TaskRepository.Delete(id);
        public async Task UpdateTask(Domain.Entities.Task updateTask) => await TaskRepository.Update(updateTask);
    }
}

File: ChallengeStefaniniGroup.UI/src/services/task/taskService.js
import * as taskResource from './taskResource';

export const getTask = () => {
    return taskResource.getTask().then(({
        data
    }) => data)
}

export const getTaskById = (id) => {
    return taskResource.getTaskById(id).then(({
        data
    }) => data)
}

export const postTask = (object) => {
    return taskResource.postTask(object).then(({
        data
    }) => data)
}

export const putTask = (object) => {
    return taskResource.putTask(object).then(({
        data
    }) => data)
}

export const deleteTask = (id) => {
    return taskResource.deleteTask(id).then(({
        data
    }) => data)
}

